File: mart/mart-domain/src/main/java/shopping/mart/domain/repository/ProductRepository.java
package shopping.mart.domain.repository;

import java.util.List;
import java.util.Optional;
import shopping.mart.domain.Product;

public interface ProductRepository {

    void saveProduct(Product product);

    Optional<Product> findByProductName(String name);

    Optional<Product> findById(long id);

    List<Product> findAllProducts();

}

File: src/main/kotlin/shopping/product/ProductRepository.kt
package shopping.product

import org.springframework.data.jpa.repository.JpaRepository

interface ProductRepository : JpaRepository<Product, Long> {
    fun getOrThrow(id: Long): Product = findById(id).orElseThrow { IllegalArgumentException("상품을 찾을 수 없습니다: $id") }
}
